src/operator/nn/pooling.cc

```diff
@@ -223,12 +223,13 @@ void PoolingComputeExCPU(const nnvm::NodeAttrs &attrs, const OpContext &ctx,
                          const std::vector<NDArray> &outputs) {
   const PoolingParam &param = nnvm::get<PoolingParam>(attrs.parsed);
   const NDArray *workspace = nullptr;
-  if (MKLDNNRequireWorkspace(param)) {
-    CHECK_GT(outputs.size(), 1U);
-    workspace = &outputs[1];
-  }
-  if (SupportMKLDNN(inputs[0])
-      && SupportMKLDNNPooling(param, inputs[0].shape())) {
+  if (SupportMKLDNN(inputs[0]) &&
+      SupportMKLDNNPooling(param, inputs[0].shape()) &&
+      SupportMKLDNNPooling(inputs[0], outputs[0])) {
```

Comment:
can you merge these two versions of `SupportMKLDNNPooling`?

Reply:
the SupportMKLDNNPooling that just accepts a param is used very extensively in other parts of the code (in GetNumOutputs, GetNumInputs, PoolingType and 15 other places). I can just rename this function.